core/src/main/java/com/expedia/adaptivealerting/core/evaluator/RmseEvaluator.java

```diff
@@ -13,16 +11,46 @@
 
     private int n;
     private double resSumSquares;
+    private double rmse;
 
     public RmseEvaluator() {
         this.n = 0;
         this.resSumSquares = 0;
+        this.rmse = 0;
     }
 
     @Override
-    public double computeScore(double observed, double predicted) {
-        this.resSumSquares += MathUtil.getSquare(observed - predicted);
+    public void update(double observed, double predicted) {
+        double residual = observed - predicted;
+        this.resSumSquares += residual * residual;
         this.n += 1;
-        return MathUtil.getSquareRoot(resSumSquares / n);
+        setRmse(Math.sqrt(resSumSquares / n));
+    }
+
+    @Override
+    public double evaluate() {
+        return rmse;
+    }
+
+    @Override
+    public void reset() {
+        this.n = 0;
+        this.resSumSquares = 0;
+    }
+
+    /**
+     * @return the rmse
+     */
+    public double getRmse() {
+        return rmse;
     }
+
+    /**
+     * @param rmse
+     *            the rmse to set
+     */
+    public void setRmse(double rmse) {
```

Comment:
As above I recommend removing this, but even if we kept RMSE as a state variable, we wouldn't want an external setter. Your `rmse` is an internal state variable so it doesn't make sense to provide an interface for external clients to set it.